azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java

```diff
@@ -146,26 +149,29 @@ public void execute() throws MojoFailureException {
         }
     }
 
-    protected void initExecution() throws MojoFailureException, MavenDecryptException, AzureExecutionException,
-        com.microsoft.azure.toolkit.lib.auth.exception.InvalidConfigurationException {
+    protected void initExecution()
+            throws MojoFailureException, MavenDecryptException, AzureExecutionException, LoginFailureException {
         // init proxy manager
         ProxyUtils.initProxy(Optional.ofNullable(this.session).map(s -> s.getRequest()).orElse(null));
         // Init telemetries
         initTelemetry();
+        Azure.az().config().setLogLevel(LogLevel.NONE);
+        Azure.az().config().setUserAgent(getUserAgent());
         trackMojoExecution(MojoStatus.Start);
         final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
         mavenAuthConfiguration.setType(getAuthType());
-        this.azureCredentialWrapper = MavenAuthManager.getInstance().login(session, settingsDecrypter, mavenAuthConfiguration);
-        if (Objects.isNull(azureCredentialWrapper)) {
+        MavenAuthManager.getInstance().login(MavenAuthManager.getInstance().buildAuthConfiguration(session, settingsDecrypter, mavenAuthConfiguration));
```

Comment:
Seems that `MavenAuthManager.login` just call the `Azure.az(AzureAccount.class).login`, so shall we using `MavenAuthManager` to get the auth configuration and call az.login here?

Reply:
The design expects you to login first and then you can get the credential anywhere